dotCMS/src/main/java/com/dotcms/vanityurl/cache/VanityUrlCacheImpl.java

```diff
@@ -0,0 +1,156 @@
+
+    @Override
+    public void remove(final Contentlet vanityURL) {
+        if (vanityURL == null || !vanityURL.isVanityUrl()) {
+            return;
+        }
+        Host host = Try.of(() -> APILocator.getHostAPI().find(vanityURL.getHost(), APILocator.systemUser(), false)).getOrNull();
+        Language lang = Try.of(() -> APILocator.getLanguageAPI().getLanguage(vanityURL.getLanguageId())).getOrNull();
+        if (host == null || lang == null) {
+            return;
+        }
+        remove(host, lang);
+
+
+    }
+
+
+    @Override
+    public void remove(final Host vanityHost) {
+        if (vanityHost == null) {
+            return;
+        }
+
+        final List<Language> langs = APILocator.getLanguageAPI().getLanguages();
+
+        for (Language lang : langs) {
+            remove(vanityHost, lang);
+        }
+
+    }
+
+
+    @Override
+    public void remove(final Host vanityHost, final Language lang) {
+        if (vanityHost == null || lang == null) {
+            return;
+        }
+
+        cache.remove(key(vanityHost, lang), VANITY_URL_SITE_GROUP);
+        cache.flushGroup(VANITY_URL_DIRECT_GROUP);
+
+    }
+
+
+    @Override
+    public void putSiteMappings(final Host host, final Language lang, final List<CachedVanityUrl> vanityURLs) {
+        if (host == null || host.getIdentifier() == null || lang == null || vanityURLs == null) {
+            return;
+        }
+        cache.put(key(host, lang), vanityURLs, VANITY_URL_SITE_GROUP);
+    }
+
+    @Override
+    public List<CachedVanityUrl> getSiteMappings(final Host host, final Language lang) {
+        if (host == null || lang == null || host.getIdentifier() == null) {
+            return null;
+        }
+        final String key = key(host, lang);
+
+        return (List<CachedVanityUrl>) cache.getNoThrow(key, VANITY_URL_SITE_GROUP);
+
+    }
+
+    @Override
+    public Optional<CachedVanityUrl> getDirectMapping(final String url, final Host host, final Language lang) {
+
+        return (Optional<CachedVanityUrl>) cache.getNoThrow(key(host, lang, url), VANITY_URL_DIRECT_GROUP);
+
+    }
+    
+
+
+
+    @Override
+    public void putDirectMapping(final Host host, final Language lang, final String url, final Optional<CachedVanityUrl> vanityUrl) {
+        if(vanityUrl==null) {
+            throw new DotRuntimeException("Unable to put null value in cache:" + key(host, lang, url));
+        }
+        cache.put(key(host, lang, url), vanityUrl, VANITY_URL_DIRECT_GROUP);
+
+    }
+
+
+    String key(final Host host, final Language lang) {
+        return key(host, lang, null);
+    }
+
+
+
+    String key(final Host host, final Language lang, final String url) {
+        if(host==null || lang==null) {
+            throw new DotRuntimeException("Host or language are null - host:" + host  + " lang:" + lang);
+        }
+        String key = host.getIdentifier() + StringPool.UNDERLINE
+                        +  String.valueOf(lang.getId()) + StringPool.UNDERLINE
```

Comment:
Codacy found an issue: [No need to call String.valueOf to append to a string.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)